ethcore/src/executive.rs

```diff
@@ -440,7 +440,11 @@ impl<'a, B: 'a + StateBackend, E: Engine + ?Sized> Executive<'a, B, E> {
 						);
 					}
 
-					Ok((params.gas - cost, ReturnData::empty()))
+					Ok(FinalizationResult {
+						gas_left: params.gas - cost,
+						return_data: ReturnData::new(output.to_owned(), 0, output.len()),
```

Comment:
isn't it dangerous to copy the output (via `to_owned`)? iirc, we've always avoided that at any cost

Reply:
There is no way around it. The built-ins don't have their own evm memory and the result has to be kept around after the call in return data buffer.